### PR TITLE
Refs #3998 Make mutationTries configurable [5018]

### PR DESCRIPTION
**Pull request by LuisGP**
Related with #324.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1399/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1368/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1379/)

File: include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h [hunk 1]
```diff
@@ -139,6 +142,7 @@ class BuiltinAttributes{
             use_WriterLivelinessProtocol = true;
             readerHistoryMemoryPolicy = MemoryManagementPolicy_t::PREALLOCATED_MEMORY_MODE;
             writerHistoryMemoryPolicy = MemoryManagementPolicy_t::PREALLOCATED_MEMORY_MODE;
+            mutation_tries = 100;
```
Comment (MiguelCompany) [inline include/fastrtps/rtps/attributes/RTPSParticipantAttributes.h]:
```suggestion
            mutation_tries = 100u;
```

File: test/unittest/xmlparser/XMLParserTests.cpp [hunk 2]
```diff
@@ -412,6 +412,7 @@ TEST_F(XMLParserTests, Data)
     EXPECT_EQ(*(loc_list_it = builtin.initialPeersList.begin()), locator);
     EXPECT_EQ(builtin.readerHistoryMemoryPolicy, PREALLOCATED_MEMORY_MODE);
     EXPECT_EQ(builtin.writerHistoryMemoryPolicy, PREALLOCATED_MEMORY_MODE);
+    EXPECT_EQ(builtin.mutation_tries, 55);
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLParserTests.cpp]:
```suggestion
    EXPECT_EQ(builtin.mutation_tries, 55u);
```

File: test/unittest/xmlparser/XMLParserTests.cpp [hunk 3]
```diff
@@ -498,6 +499,7 @@ TEST_F(XMLParserTests, DataBuffer)
...
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLParserTests.cpp]:
```suggestion
    EXPECT_EQ(builtin.mutation_tries, 55u);
```

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 4]
```diff
@@ -124,6 +124,7 @@ TEST_F(XMLProfileParserTests, XMLParserParcipant)
     EXPECT_EQ(*(loc_list_it = builtin.initialPeersList.begin()), locator);
     EXPECT_EQ(builtin.readerHistoryMemoryPolicy, PREALLOCATED_MEMORY_MODE);
     EXPECT_EQ(builtin.writerHistoryMemoryPolicy, PREALLOCATED_MEMORY_MODE);
+    EXPECT_EQ(builtin.mutation_tries, 55);
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp]:
```suggestion
    EXPECT_EQ(builtin.mutation_tries, 55u);
```

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 5]
```diff
@@ -192,6 +193,7 @@ TEST_F(XMLProfileParserTests, XMLParserDefaultParcipantProfile)
...
```
Comment (MiguelCompany) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp]:
```suggestion
    EXPECT_EQ(builtin.mutation_tries, 55u);
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1403/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1372/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1383/)

**Pull request merged**